src/features/defi/providers/osmosis/components/OsmosisManager/Deposit/OsmosisDeposit.tsx

```diff
@@ -0,0 +1,193 @@
+} from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { Confirm } from './components/Confirm'
+import { Deposit } from './components/Deposit'
+import { Status } from './components/Status'
+import { OsmosisDepositActionType } from './DepositCommon'
+import { DepositContext } from './DepositContext'
+import { initialState, reducer } from './DepositReducer'
+
+type OsmosisDepositProps = {
+  accountId: AccountId | undefined
+  onAccountIdChange: AccountDropdownProps['onChange']
+}
+
+export const OsmosisDeposit: React.FC<OsmosisDepositProps> = ({
+  accountId,
+  onAccountIdChange: handleAccountIdChange,
+}) => {
+  const [state, dispatch] = useReducer(reducer, initialState)
+  const translate = useTranslate()
+  const { query, history, location } = useBrowserRouter<DefiQueryParams, DefiParams>()
+  const { chainId, assetNamespace, assetReference } = query
+  const userAddress = useMemo(() => accountId && fromAccountId(accountId).account, [accountId])
+
+  const assetId = toAssetId({
+    chainId,
+    assetNamespace,
+    assetReference,
+  })
+
+  const asset = useAppSelector(state => selectAssetById(state, assetId))
+  const loading = useSelector(selectPortfolioLoading)
+
+  const opportunityId: LpId | undefined = useMemo(
+    () => (assetId ? toOpportunityId({ chainId, assetNamespace, assetReference }) : undefined),
+    [assetId, assetNamespace, assetReference, chainId],
+  )
+
+  const highestBalanceAccountIdFilter = useMemo(
+    () => ({ stakingId: opportunityId }),
+    [opportunityId],
+  )
+  const highestBalanceAccountId = useAppSelector(state =>
+    selectHighestBalanceAccountIdByStakingId(state, highestBalanceAccountIdFilter),
+  )
+
+  const OsmosisLpOpportunityFilter = useMemo(
+    () => ({
+      lpId: opportunityId ?? '',
+      assetId,
+      accountId: highestBalanceAccountId ?? '',
+    }),
+    [assetId, highestBalanceAccountId, opportunityId],
+  )
+  const opportunity = useAppSelector(state =>
+    selectEarnUserLpOpportunity(state, OsmosisLpOpportunityFilter),
+  )
+
+  const underlyingAsset0Id = useMemo(
+    () => opportunity?.underlyingAssetIds[0] ?? '',
+    [opportunity?.underlyingAssetIds],
+  )
+  const underlyingAsset1Id = useMemo(
+    () => opportunity?.underlyingAssetIds[1] ?? '',
+    [opportunity?.underlyingAssetIds],
+  )
+
+  const underlyingAsset0: Asset | undefined = useAppSelector(state =>
+    selectAssetById(state, underlyingAsset0Id),
+  )
+  const underlyingAsset1: Asset | undefined = useAppSelector(state =>
+    selectAssetById(state, underlyingAsset1Id),
+  )
+
+  const getPoolData = useCallback(async (): Promise<OsmosisPool | undefined> => {
+    if (!opportunity) return undefined
+    const opportunityAssetId = opportunity.assetId
+    if (!opportunityAssetId) return
+    const { assetReference: poolAssetReference } = fromAssetId(opportunityAssetId)
+    if (!poolAssetReference) return
+    const id = getPoolIdFromAssetReference(poolAssetReference)
```

Comment:
If we're able to get the pool ID so easily, I'd probably pass CAIPs everywhere vs. Osmosis pool IDs and do the conversion in-place to keep things tidier